### DISCUSSION TITLE
Weather forecast for next race

### DISCUSSION THREAD

**Discussion by vlx** [marcokreeft87/formulaone-card#145] [Ideas]
What about adding weather info for the days of the next race card? It might require additional data source and I don't know how hard is to implement. But I think it will add more to the suspense.

**Comment by marcokreeft87** [comment]
Yes I like this one! It's going on the todo

**Comment by ndrinta** [comment]
Hi, just tried this feature and I love it!
Just one thing, what is the difference between the 2 temperatures?
![image](https://user-images.githubusercontent.com/4704172/223687003-5176d108-5711-44f5-b662-631c1f32eddb.png)
Cheers

**Comment by marcokreeft87** [comment]
Temperature and feels like

**Comment by ArtMel1998** [comment]
Beuatifull feature for HA. I just love it. Since this today (07-04-2023) the weather forecast is not working for me any more

HA version 2023.4.1
Supervisor 2023.04.0
OS 9.5
Frontend 20230406.1
Formula1 card 1.4.2

Issue: weather and race-info not visible when show_weather has been added correctly to formula1 card (next_race). When weather info is not added race info is shown. 
It worked before this weeks update.

code without weather_info:

`type: custom:formulaone-card
card_type: next_race
date_locale: NL
image_clickable: false
show_raceinfo: true`

code with weather_info:

`type: custom:formulaone-card
card_type: next_race
date_locale: NL
image_clickable: false
show_raceinfo: true
show_weather: true
weather_options:
  unit: metric
  api_key: [my api key]`

**Reply by marcokreeft87** [reply]
Check v1.4.3

**Comment by ArtMel1998** [comment]
solved! Happy Eastern

**Comment by GroteGehaktBal** [comment]
When creating the API Key on visualcrossing, does it matter which location I fill in to create the API key?

> All requests must include a location. A location can be an address, partial address (eg zip code or city, country) or latitude,longitude (eg 52.06,-2.72)

**Reply by marcokreeft87** [reply]
Nah not really. Just take Amsterdam or something

**Comment by JohnHasselhof** [comment]
I wanted to use the f1sensor. However when entering the following code, no weather is shown:

type: custom:formulaone-card
show_weather: true
weather_options:
  source: f1sensor
  unit: metric
  
 What do I do wrong?

**Reply by marcokreeft87** [reply]
You are missing the api key

**Reply by JohnHasselhof** [reply]
ah, so I still need an api key even f1sensor shows me the actual weather conditions in sensor.f1_weather?

**Reply by marcokreeft87** [reply]
> ah, so I still need an api key even f1sensor shows me the actual weather conditions in sensor.f1_weather?
> 

Oh sorry. I missed that. The type of f1 card isn't in your config

Card_type

**Reply by JohnHasselhof** [reply]
Thanks! But if I put in card_type: next_race it only shows the course and no weather.

**Reply by marcokreeft87** [reply]
Try show_raceinfo: true

**Reply by JohnHasselhof** [reply]
No, this shows all information about race, but weather is lacking.

**Reply by marcokreeft87** [reply]
I see now.

Entity is missing

This is my config

```type: custom:formulaone-card
    card_type: countdown
    show_raceinfo: true
    source: f1sensor
    entity: sensor.f1_current_season
    f1_font: true
    date_locale: en-US
    show_weather: true
    weather_options:
      source: f1sensor
      entity: sensor.f1_weather
    title: null
    show_lastyears_result: true
    hide_tracklayout: false
    show_refresh: false

**Reply by JohnHasselhof** [reply]
Hm, strange. Using exactly this config by copy/paste, it shows everything but the not the weather.

**Reply by marcokreeft87** [reply]
Is the sensor giving data?

**Reply by oOStroudyOo** [reply]
I'm also having the same problem:
![image](https://github.com/user-attachments/assets/f0da552e-e164-4745-98d5-15f87923a16d)

```
type: custom:formulaone-card
card_type: countdown
show_raceinfo: true
f1_font: true
date_locale: en-US
show_weather: true
weather_options:
  source: f1sensor
  entity: sensor.f1_weather
title: null
show_lastyears_result: false
hide_tracklayout: false
show_refresh: false
```

![image](https://github.com/user-attachments/assets/7596f74a-d3bc-44f3-8525-82558293e822)

**Reply by marcokreeft87** [reply]
Why do you use the f1 sensor for the weather but not the countdown?

**Reply by JohnHasselhof** [reply]
Thanks, it's exactly the same pictures for me.

**Reply by marcokreeft87** [reply]
Yeah all data is null from the sensor.
The sensor doesn't show any data of the race weather. Try asking at the github of the sensor

**Reply by oOStroudyOo** [reply]
> Why do you use the f1 sensor for the weather but not the countdown?

Because the API key from the other source was problematic, I found myself needing to change my key frequently to get it to work again. Everything else worked so left it as default.

Eitherway, even if I copy and paste your config from above, I don't get any weather:
![image](https://github.com/user-attachments/assets/be339ee0-9535-412e-bab9-1566c28e73b0)

**Reply by marcokreeft87** [reply]
Ah I see. But the f1 sensor is at fault here unfortunately. I can't do anything about that

**Reply by oOStroudyOo** [reply]
Issue raised:
https://github.com/Nicxe/f1_sensor/issues/8

**Reply by oOStroudyOo** [reply]
Hi @marcokreeft87 

The sensor does have attributes now, but I still see nothing on my card, could you explain if I've done something wrong in my config please:
![image](https://github.com/user-attachments/assets/555ac446-f41b-4c10-a903-e260cf381626)

**Reply by marcokreeft87** [reply]
> Hi @marcokreeft87 
> 
> The sensor does have attributes now, but I still see nothing on my card, could you explain if I've done something wrong in my config please:
> ![image](https://github.com/user-attachments/assets/555ac446-f41b-4c10-a903-e260cf381626)

Compare it again with my example above

**Reply by oOStroudyOo** [reply]
@marcokreeft87 

Copied and pasted your config:
![image](https://github.com/user-attachments/assets/7a13202c-c0b7-40d9-ac77-db9a708e7814)

**Reply by marcokreeft87** [reply]
Ah apparently it still needs an api key in the Configuration. Weird, just fill in something doest need to be a valid one

**Reply by oOStroudyOo** [reply]
You're right @marcokreeft87 

Infact it doesn't even need an entry, it just needs to exist.
This works fine:

```
type: custom:formulaone-card
card_type: countdown
show_raceinfo: true
source: f1sensor
entity: sensor.f1_current_season
f1_font: true
date_locale: en-US
show_weather: true
weather_options:
  source: f1sensor
  entity: sensor.f1_weather
  api_key:
title: null
show_lastyears_result: true
hide_tracklayout: false
show_refresh: false
```

Thank you for your patience.

**Reply by JohnHasselhof** [reply]
Thanks both of you! Adding the empty api_key is working for me as well.
